src/socket.rs

```diff
@@ -382,3 +418,65 @@ impl From<mpsc::error::SendError<StateCommand>> for DisconnectError {
         DisconnectError::SocketShutdown
     }
 }
+
+#[derive(Clone)]
+struct ArcStatus {
```

Comment:
ArcStatus isn't a good name for this; maybe ObservableStatus?

Reply:
I thought of using Observable in the name, but I couldn't remember if Observable was a term used in Rust as I only remembered it in Java/Kotlin and JavaScript.  Thanks for the confirmation the name makes sense in Rust too.  ❤️